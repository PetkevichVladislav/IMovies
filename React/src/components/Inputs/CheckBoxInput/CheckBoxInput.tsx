import { FC } from "react";

import "./CheckBoxInput.scss";
import "../Input.scss";

export interface ICheckBoxInput {
    label: string;
    isChecked?: boolean;
    onChange: (event : React.ChangeEvent<HTMLInputElement>) => void;
}

export const CheckBoxInput: FC<ICheckBoxInput> = ({ isChecked = false, label, onChange }) => {
    return (
        <label className="checkbox-input__container" data-testid="checkbox-container">{label}
            <input className="checkbox-input__input"
                key={label.replace(" ", "-").toLowerCase()}
                value={label}
                type="checkbox"
                checked={isChecked}
                onChange={event => onChange(event)}
                data-testid="checkbox-input" />
            <span className="checkbox-input__checkmark"></span>
        </label>
    )
}