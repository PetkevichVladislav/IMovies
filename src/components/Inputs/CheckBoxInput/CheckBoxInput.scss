.checkbox-input {
    &__container {
        display: block;
        position: relative;
        padding-left: 23px;
        cursor: pointer;
        font-size: 20px;
        -webkit-user-select: none;
        -moz-user-select: none;
        -ms-user-select: none;
        user-select: none;
        color: #FFFFFF;

        &:hover input~.checkbox-input__checkmark {
            background-color: #E44C5A;
        }

        input:checked~.checkbox-input__checkmark:after {
            display: block;
        }
    }

    &__input {
        position: absolute;
        opacity: 0;
        cursor: pointer;
        height: 0;
        width: 0;
        

        &:checked~.checkbox-input__checkmark {
            background-color: #E44C5A;
        }
    }

    &__checkmark {
        position: absolute;
        top: 4px;
        left: 0px;
        height: 16px;
        width: 16px;
        border-radius: 2px;
        background-color: #eee;

        &:after {
            content: "";
            position: absolute;
            display: none;
            left: 5px;
            top: 2px;
            width: 4px;
            height: 8px;
            border: solid white;
            border-width: 0 2px 2px 0;
            transform: rotate(45deg);
        }
    }
}