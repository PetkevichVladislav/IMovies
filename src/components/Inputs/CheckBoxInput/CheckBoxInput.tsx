import { FC, useCallback, useState } from "react";

import "./CheckBoxInput.scss";
import "../Input.scss";

export interface ICheckBoxInput {
    label: string;
    defaultChecked?: boolean;
    name?: string;
    onChange: (isChecked: boolean) => void;
}

export const CheckBoxInput: FC<ICheckBoxInput> = ({defaultChecked = false, label, onChange, name}) => {
    const [isChecked, setIsChecked] = useState(defaultChecked);

    const onChanged = useCallback(() => {
        setIsChecked(!isChecked);
        onChange(!isChecked);
    }, [onChange, isChecked, setIsChecked]);

    return (
        <>
            <label className="checkbox-input__container" defaultChecked={isChecked} onChange={onChanged}>{label}
                <input className="checkbox-input__input" name={name} type="checkbox" checked={isChecked}/>
                <span className="checkbox-input__checkmark"></span>
            </label>
        </>
    )
}