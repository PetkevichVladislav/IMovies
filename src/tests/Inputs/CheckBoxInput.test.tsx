import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { CheckBoxInput, ICheckBoxInput } from '../../components/Inputs/CheckBoxInput/CheckBoxInput';

describe("Date time Input", () => {
    const onChangeCallback = jest.fn();
    const textInputData : ICheckBoxInput = {
        label: "check if you junior",
        name: "checker",
        onChange: onChangeCallback,
    };

    it("When data is valid component should renders correctly", () =>{
        const tree = renderer.create(<CheckBoxInput {...textInputData}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});