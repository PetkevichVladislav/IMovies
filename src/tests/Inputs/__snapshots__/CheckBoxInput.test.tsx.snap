// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date time Input When data is valid component should renders correctly 1`] = `
<label
  className="checkbox-input__container"
  data-testid="checkbox-container"
  defaultChecked={false}
  onChange={[Function]}
>
  check if you junior
  <input
    checked={false}
    className="checkbox-input__input"
    data-testid="checkbox-input"
    name="checker"
    type="checkbox"
  />
  <span
    className="checkbox-input__checkmark"
  />
</label>
`;
